Assets/Shopify/UIToolkit/Editor/ShopCredentialsVerifier.cs

```diff
@@ -124,8 +131,12 @@ public class ShopCredentialsVerifier {
             }
         }
 
-        private ShopifyClient Client() {
-            return new ShopifyClient(new UnityEditorLoader(_Credentials.ShopDomain, _Credentials.AccessToken));
+        private ShopifyClient MakeClient() {
```

Comment:
If we pass in a blank domain, when we go to contruct a URL inside UnityLoader it blows up. Before continuing we check to see if it was empty/blank string and return null otherwise so I've renamed this method to `MakeClient`.

Reply:
This only fixes the issue here, should this check be inside the ShopifyClient constructor, and raise an exception if the constructor arguments are not valid?

Reply:
Definitely. I've updated it so the constructor throws and we catch in the `ShopCredentialsVerifier` if it's invalid and update our state accordingly.